### PR TITLE
Add some vocab

### PR DESCRIPTION
**Pull request by lpercival**
How does this look? Do we still need the TODO?

### CODE REVIEW

File: resources/vocabulary.md [hunk 1]
```diff
@@ -93,8 +93,18 @@ sum_of_nums_under(4) // 10
 sum_of_nums_under(6) // 21
 ```
 
+## Process and Tools
```
Comment (helin24) [inline resources/vocabulary.md]: Kind of unrelated, but do you want to change the two main sections of this page to ### Things most programming languages do  and ### Process and tools (like delete line 3)? The outline right now is a little weird.

File: resources/vocabulary.md [hunk 2]
```diff
@@ -93,8 +93,18 @@ sum_of_nums_under(4) // 10
...
+- IDE: Integrated Development Environment - a piece of software that you write code in, typically with helpful features like tab completion, compilation, and debugging
```
Comment (helin24) [inline resources/vocabulary.md]: Maybe mention an example? IntelliJ for one.

File: resources/vocabulary.md [hunk 3]
```diff
@@ -93,8 +93,18 @@ sum_of_nums_under(4) // 10
...
+- Algorithm: a step by step sequence of events to accomplish a goal, like a recipe.
+- Pseudocode: a helpful step in the development process of writing out what you plan to do in regular language before coding it.
+- Framework: A scaffold that you can build applications around. Includes reusable code that provides a set of basic, generic functionality for you to build upon. A common example is MVC (Model-View-Controller) frameworks for building web apps, such as Django.
+- Library/ Package: A set of resources that someone else has already written and provided so that you can make use of them rather than having to write that code yourself. An example would be the string library in C.
+- REPL: Stands for Read-Eval-Print-Loop. Simple programming environment that acts as a language shell and lets us write and test code without having to instal and configure a bunch of prerequisites.
```
Comment (helin24) [inline resources/vocabulary.md]: Maybe delete 'acts as a language shell' which may be more confusing than the later explanation. +L on install

**Review by helin24 — APPROVED** [review]

File: resources/vocabulary.md [hunk 4]
```diff
@@ -93,8 +93,18 @@ sum_of_nums_under(4) // 10
...
+- CLI: Command Line Interface. Interacting with your computer via text commands in the shell.
+
+
 ### TODO(lpercival): add other vocabulary
 - Specific words from descriptions of common concepts?
```
Comment (helin24) [inline resources/vocabulary.md]: Hmm.. I totally don't remember what I meant here. Maybe just delete this part now?

File: resources/vocabulary.md (hunk 1 above)
Comment (lpercival) [reply resources/vocabulary.md]: I think we decided to split into 2 files as part of the re-org, so I'll let this be for the moment

**Pull request merged**